lib/tool_shed/galaxy_install/tool_dependencies/recipe/tag_handler.py

```diff
@@ -466,6 +466,10 @@ def handle_complex_repository_dependency_for_package( self, elem, package_name,
                     if not os.path.exists( required_repository_package_install_dir ):
                         log.error( 'Missing required tool dependency directory %s' % str( required_repository_package_install_dir ) )
                     repo_files_dir = required_repository.repo_files_directory( self.app )
+                    if not repo_files_dir:
+                        message = "Unable to locate the repository directory for revision %s of installed repository %s owned by %s." % \
+                            ( str( required_repository.changeset_revision ), str( required_repository.name ), str( required_repository.owner ) )
```

Comment:
I don't think you need the extra str() casts here. `%s` is taking care of it already.